### PR TITLE
Update Circuit Breaker configured as a standard plugin

### PR DESCRIPTION
**Pull request by atris**
This commit refactors circuit breaker configurations to use PluginInfo and reduce the footprint on SolrConfig.

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/core/SolrCore.java [hunk 1]
```diff
@@ -1174,7 +1174,8 @@ private SolrCoreMetricManager initCoreMetricManager(SolrConfig config) {
   }
 
   private CircuitBreakerManager initCircuitBreakerManager() {
-    CircuitBreakerManager circuitBreakerManager = CircuitBreakerManager.build(solrConfig);
+    final PluginInfo info = solrConfig.getPluginInfo(CircuitBreakerManager.class.getName());
+    CircuitBreakerManager circuitBreakerManager = CircuitBreakerManager.build(info.initArgs);
```
Comment (noblepaul) [inline solr/core/src/java/org/apache/solr/core/SolrCore.java:1178]: Make `CircuitBreakerManager` implement `PluginInfoInitialized`

File: solr/solr-ref-guide/src/query-settings-in-solrconfig.adoc [hunk 2]
```diff
@@ -172,24 +172,54 @@ This parameter sets the maximum number of documents to cache for any entry in th
...
+  <!-- All specific configs in this section -->
+</circuitBreaker>
 ----
 
-=== memoryCircuitBreakerThresholdPct
+To control whether Circuit Breakers are globally enabled, use the following flag:
+
+[source,xml]
+----
+<str name="enabled">true</str>
```
Comment (noblepaul) [inline solr/solr-ref-guide/src/query-settings-in-solrconfig.adoc:190]: standard syntax for enabled is 

`<circuitBreaker class="solr.CircuitBreakerManager" enabled="true">`

**Review by noblepaul — COMMENTED** [review]
The config looks awesome now.

File: solr/core/src/java/org/apache/solr/util/circuitbreaker/CircuitBreakerManager.java [hunk 3]
```diff
@@ -121,19 +135,41 @@ public static String toErrorMessage(List<CircuitBreaker> circuitBreakerList) {
    *
    * Any default circuit breakers should be registered here.
    */
-  public static CircuitBreakerManager build(SolrConfig solrConfig) {
-    CircuitBreakerManager circuitBreakerManager = new CircuitBreakerManager(solrConfig.useCircuitBreakers);
-
-    // Install the default circuit breakers
-    CircuitBreaker memoryCircuitBreaker = new MemoryCircuitBreaker(solrConfig);
-    CircuitBreaker cpuCircuitBreaker = new CPUCircuitBreaker(solrConfig);
+  @SuppressWarnings({"rawtypes"})
```
Comment (madrob) [inline solr/core/src/java/org/apache/solr/util/circuitbreaker/CircuitBreakerManager.java:138]: can we avoid adding new suppress to the code please?

Comment (atris) [reply solr/core/src/java/org/apache/solr/util/circuitbreaker/CircuitBreakerManager.java:138]: Unfortunately, that is the way NamedArgs are structured in PluginInfo -- we need to fix that before we remove this warning. I will follow up.

**Pull request merged**